scripts/build_demand_profiles.py

```diff
@@ -106,16 +107,41 @@ def get_load_paths_gegis(ssp_parentfolder, config):
     prediction_year = config.get("load_options")["prediction_year"]
     ssp = config.get("load_options")["ssp"]
 
+    scenario_path = os.path.join(ssp_parentfolder, ssp)
+
     load_paths = []
-    for continent in region_load:
-        load_path = os.path.join(
-            ssp_parentfolder,
-            str(ssp),
-            str(prediction_year),
-            "era5_" + str(weather_year),
-            str(continent) + ".nc",
-        )
-        load_paths.append(load_path)
+    load_dir = os.path.join(
+        ssp_parentfolder,
+        str(ssp),
+        str(prediction_year),
+        "era5_" + str(weather_year),
+    )
+
+    if os.path.exists(scenario_path):
+        for continent in region_load:
+            for ext in [".nc", ".csv"]:
+                load_path = os.path.join(str(load_dir), str(continent) + str(ext))
+                if os.path.exists(load_path):
+                    load_paths.append(load_path)
+                    break
+
+        avail_regions = [
+            os.path.split(os.path.abspath(pth))[1].split(".nc")[0] for pth in load_paths
+        ]
+
+        logger.info(f" An assumed load folder {load_dir}, load path is {load_paths}.")
+
+        if len(load_paths) == 0:
+            logger.warning(
+                f"No demand data file for {set(region_load).difference(avail_regions)}. An assumed load folder {load_dir}."
+            )
```

Comment:
What if we drop the exists of scenario path and we do:
```suggestion
        regions_found = []
        file_names = []
        for continent in region_load:
            sel_ext = ".nc"
            for ext in [".nc", ".csv"]:
                load_path = os.path.join(str(load_dir), str(continent) + str(ext))
                if os.path.exists(load_path):
                    sel_ext = ext
                    regions_found.append(continent)
                    break
             file_name = str(continent) + str(sel_ext)
             load_path = os.path.join(str(load_dir), file_name)
             load_paths.append(load_path)
             file_names.append(file_name)
        
        logger.info(
            f"Demand data folder: {load_dir}, load path is {load_paths}.\n" +
            f"Expected files: " + "; ".join(file_names)
        )
```
If the file does not exist, this solution should backup to the .nc case.
Not sure about how much info to disclose;
I'm unsure it is good to advice about whether demand data files are existing or not: for the standard user that may be confusing because of the ordering about when snakemake makes this evaluation.

What do you think?

Reply:
I love the solution! 😄 Implemented. Thanks a lot for that.

Agree that the no-demand warning feels redundant. Moreover, it's also not needed: in case no demand data provided and `retrive_databundle: false`, Snakemake just throws an error even before running anything. So, we can just remove this checking part.